### ISSUE TITLE
Error TS2503: Cannot find namespace 'Microsoft'

### ISSUE THREAD

**Issue by sjdrew** [youjustgo/ng2-bingmaps#32]
Trying to use in an ionic2/angular2 app and typescript does not compile.

I added the import:
import { NG2_BINGMAPS_PROVIDERS, LazyMapsAPILoaderConfig} from 'ng2-bingmaps/core';

And the providers as documented and then app does not compile. Any where Microsoft.Maps global namespace is referenced it triggers the compiler error.  For example:

TypeScript error: myapp/node_modules/ng2-bingmaps/directives/bing-map.d.ts(103,16): Error TS2503: Cannot find namespace 'Microsoft'.
TypeScript error: myapp/node_modules/ng2-bingmaps/directives/bing-map.d.ts(104,35): Error TS2503: Cannot find namespace 'Microsoft'.

Thanks.
Steve.

**Comment by rolandoldengarm** [comment]
What TypeScript version do you have?

**Comment by sjdrew** [comment]
tsc Version 1.8.10

**Comment by rolandoldengarm** [comment]
Hmm, that should be fine.
Could you please try and add the typing "bingmaps/microsoft.maps"
I know, it should not be necessary, just trying if that works for you. I have only used this in projects where we also used the native maps API.

**Comment by sjdrew** [comment]
I had tried that already with no change.

 

I did this, in the project root folder: 

 

typings install --save --global dt~bingmaps/microsoft.maps

**Comment by omeryounus** [comment]
I am having the same issue, it seems like it imported in the build process? any solution?

**Comment by rolandoldengarm** [comment]
It's weird that I cannot reproduce it, using it in 2 different projects. Maybe we're doing something wrong with our package creation / build process? 
Are you able to reproduce it in a plunkr?

**Comment by omeryounus** [comment]
I have checked on Webstorm and Visual Studio but not able to fix the issue. It says 'cannot find namespace Microsoft.maps....' I have used 'npm install --save ng2-bingmaps' into the project but nothing changed.

**Comment by rolandoldengarm** [comment]
Just out of curiosity, have you tried running 
"typings install --global dt~bingmaps/microsoft.maps" ?
to install the tsd's to your project?

**Comment by omeryounus** [comment]
Yes, I tried to install the same but nothing appears to happened. Still having same issue.

**Comment by rolandoldengarm** [comment]
And just to double check, you reference that tsd file in your ts code (e.g. your bootstrap component)?
E.g. /// <reference path="../typings/index.d.ts" />

**Comment by rolandoldengarm** [comment]
@omeryounus @sjdrew Please have a look at my latest comment, just double checking, are you referencing the index.d.ts file from your code?

**Comment by sjdrew** [comment]
Not specifically, in fact I do not even see a index.d.ts file within the node_modules/ng2-bingmaps  tree.

 

Steve.

**Comment by rolandoldengarm** [comment]
I mean that you need to reference the index.d.ts from your typings folder...
So 
1) typings install --global dt~bingmaps/microsoft.maps
2) reference the index.d.ts in your code from typings folder
@sjdrew

**Comment by omeryounus** [comment]
Amazing! it's working absolutely fine now. Thanks!

**Comment by rolandoldengarm** [comment]
Cool, although we need to find a permanent solution. I'll keep the issue open.

**Comment by Armorgadden** [comment]
how to reference the index.d.ts file???in which file??

**Comment by rolandoldengarm** [comment]
@Armorgadden the index.d.ts is in your typings folder.
